radio/wi-fi/wps.md

```diff
@@ -1,10 +1,132 @@
-# 🛠️ WPS
+# WPS
 
-## Recon
+## Theory
+
+Wi-Fi Protected Setup (WPS) is a simplified configuration protocol designed to make connecting devices to a secure wireless network easier.
+WPS employs an 8-digit PIN for user network connection, verifying the first 4 digits before proceeding to check the remaining 4. This allows for a potential Brute-Force attack on the first set of digits followed by the second set, with a total of only 11,000 possible combinations.
+
+There are three main methods for setting up a connection using WPS :
+
+* **Push Button Configuration (PBC) :** Press the WPS button on the router and then activate the WPS function on the device to connect. The two devices will automatically establish a connection.
+
+* **PIN-based Configuration :** Each WPS-enabled device has a unique PIN code. It can be entered on the router through its web interface, or vice versa, to establish the connection.
+
+* **Near-field communication Configuration :** In which the user has to bring the new client close to the access point to allow a near field communication between the devices. NFC Forum–compliant RFID tags can also be used.
 
 ## Attacks
 
-### Reaver
+Here are some known attacks on Wi-Fi Protected Setup (WPS) :
+
+* **PIN Brute-Force Attack :** Some routers with WPS allow users to connect by entering an 8-digit PIN code. A brute-force attack involves trying all possible combinations until the correct PIN is found.
+
+* **Pixie Dust Attack :** This attack exploits a weakness in the generation of WPS encryption keys. It involves exploiting vulnerabilities in certain routers during the creation of WPA/WPA2 encryption keys.
+
+* **PIN Revelation Attack :** Certain WPS-enabled routers may reveal information about the validity of the PIN during a connection attempt. This information disclosure can aid an attacker in narrowing down the search space during a brute-force attack.
```

Comment:
from how many possibilities to how many? Any idea of the time it takes?

Reply:
It will depend on the victim router. i can't give a time estimate.

Reply:
Fair enough, resolving this conversation 👍